lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,59 @@
+require_relative "publish_options"
+
+begin  # optional dependency
+  require 'resque'
+  require 'resque-retry'
+rescue LoadError
+end
+
+module Pwwka
+  # Primary interface for sending messages.
+  #
+  # Example:
+  #
+  #     # Send a message, blowing up if there's any problem
+  #     Pwwka::PersistentTransmitter.batch do |transmitter|
+  #         transmitter.send_message!({ user_id: @user.id }, "users.user.activated")
+  #     end  
+
+  class PersistentTransmitter
+
+    extend Pwwka::Logging
+    include Pwwka::Logging
+
+    DEFAULT_DELAY_BY_MS = 5000
+
+    attr_reader :channel_connector
+   
+    def initialize
+      @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
+    end
+
+    def send_message!(payload, routing_key, type: nil, headers: nil, message_id: :auto_generate)
+      publish_options = Pwwka::PublishOptions.new(
+        routing_key: routing_key,
+        message_id: message_id,
+        type: type,
+        headers: headers
+      )
+      logf "START Transmitting Message on id[%{id}] %{routing_key} -> %{payload}", id: publish_options.message_id, routing_key: routing_key, payload: payload
+      channel_connector.topic_exchange.publish(payload.to_json, publish_options.to_h)
```

Comment:
Is it intentional to exclude the resque/async error handling strategy found in the default transmitter?

Reply:
Only for the RC I think.  Ideas on how to handle failures are welcome.  Options I have considered so far:
1. Just let it raise an error and republish the entire batch again on retry, some being duplicates
1. After encountering an error, enqueue a job per message for remaining messages, forgoing the benefits of the batching/connection-reuse in the event of edge cases like this
1. After encountering an error, enqueue a job that contains all required information needed to publish the remaining messages in a batch
1. Go into a loop attempting to reconnect and eventually failing over to one of the above options after a period of time

Reply:
The least change would be to handle each failure individually like we do now.

Currently, if we had this:

```ruby
send_message!(...)
send_message!(...)
send_message!(...)
send_message!(...)
```

And suppose #3 failed, we would queue job to send that later, then proceed to #4.  If #4 succeeded, cool.

If this new change maintains that semantic then we can easily take any call to `send_message!` and wrap it in the new class without having to think through new failure modes.

We could achieve this pretty easily (I think!) if this class deferred to the existing `Transmitter` entirely, but handed it the connection to use (plus told it not to close it)

Reply:
> We could achieve this pretty easily (I think!) if this class deferred to the existing Transmitter entirely, but handed it the connection to use (plus told it not to close it)

I like this idea.